### PR TITLE
Drop Node.js 6 & 8  and use 10 by default

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

 - Update travis to run only tests with Node.js 10
 - replace `for (i = 0; ...` by `for (let i = 0; ...`
 - update plugins to branch kuzzle-v2 to update the manifest minimum version
 - update npm dependencies (`ws` => 7.x)
 - drop proxy tests

NB: Tests didn't pass because of the documentation that is added on a separate PR.

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L972](https://github.com/kuzzleio/kuzzle/blob/3e44a1d71a27ff44c04c863cc43f6d19d4b4b163/lib/api/core/plugins/pluginsManager.js#L972): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/c5622ffd74c681fc002dee5b6a74d5abc9750924/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/f4066c877dc31facd509c3760a7ed15c661f1a44/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/83da46555de0524d7049374a81d45e81047cf779/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/263df19ef7e76be2e813506722c2cadae8f0417f/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       cache: npm
 
       node_js:
-        - "8"
+        - "10"
```
Comment (alexandrebouthinon) [inline .travis.yml]: NP: If there is only one element you can remove list and use standard yaml field

**Comment by Aschen** [comment]
> This Pull Request is polluted by new changes on master. You should merge 1-dev with 2-dev to get the recent changes. At the moment, code review is difficult.

I already merge 1-dev back to 2-dev, I didn't understand why Github still show the diff ..

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/0da8b844fbc91619f66535cc3398dba2999e3b55/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by alexandrebouthinon** [comment]
You forget to update this branch with the new 2-dev changes

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L967](https://github.com/kuzzleio/kuzzle/blob/4ea121beee5a49ee8b24042329b5d4bd1ebf428d/lib/api/core/plugins/pluginsManager.js#L967): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by Aschen** [comment]
Tests didn't pass because of the documentation that is added on a separate PR.

File: .ci/test-aarch64.yml [hunk 2]
```diff
@@ -1,13 +1,8 @@
 version: '3'
 
 services:
-  proxy:
```
Comment (scottinet) [inline .ci/test-aarch64.yml]: :tada: :tada: :tada:

**Pull request merged**